lib/services/azuresqldb/client.js

```diff
@@ -21,56 +21,86 @@ var sqldbOperations = function (log, azure) {
 
     log.info('sqldb client CTOR');
 
+    function myRequest(url, qs, method, headers, form, json, next) {
+
+        var requestObject = { url: url, qs: qs, method: method, headers: headers, form: form, json: json };
+        async.waterfall([
+            function (callback) {
+                request(requestObject, function (err, res, body) {
+                    if (err) {
+                        log.error('sqldb client: error in REST request first attempt: %j', err);
+                        callback(null, true, res, body);
+                    } else {
+                        callback(null, false, res, body);
+                    }
+                });
+            },
+            function (retry, response, prevBody, callback) {
+                if (retry) {
+                    request(requestObject, function (err, res, body) {
+                        if (err) {
+                            log.error('sqldb client: error in REST request retry 1: %j', err);
+                            callback(null, true, res, body);
+                        } else {
+                            callback(null, false, res, body);
+                        }
+                    });
+                } else {
+                    callback(null, false, response, prevBody);
+                }
+            },
+            function (retry, response, prevBody, callback) {
+                if (retry) {
+                    request(requestObject, function (err, res, body) {
+                        if (err) {
+                            log.error('sqldb client: error in REST request retry 2: %j', err);
+                            callback(err, res, body);
+                        } else {
+                            callback(null, res, body);
+                        }
+                    });
+                } else {
+                    callback(null, response, prevBody);
+                }
+            }
+        ], function (err, response, body) {
+            next(err, response, body);
+        });
+
+    }
+
```

Comment:
In current way, if we need to retry 5 times, we need to write 5 functions in `waterfall`. Do you have another way to retry the request instead of hardcoding the retry time?